finality-aleph/src/new_network/substrate.rs

```diff
@@ -33,19 +50,14 @@ impl<B: Block, H: ExHashT> Network for Arc<NetworkService<B, H>> {
         data: impl Into<Vec<u8>> + Send + Sync + 'static,
         peer_id: PeerId,
         protocol: Cow<'static, str>,
-    ) -> Result<(), NetworkSendError> {
-        if let Ok(sender) = self.notification_sender(peer_id.into(), protocol) {
-            sender
-                .ready()
-                .await
-                .ok()
-                .unwrap()
-                .send(data)
-                .map_err(|_| NetworkSendError::Closed)
-        } else {
-            debug!(target: "aleph-network", "Attempted send to peer we are not connected to.");
-            Err(NetworkSendError::Closed)
-        }
+    ) -> Result<(), SendError> {
+        self.notification_sender(peer_id.into(), protocol)
+            .map_err(|_| SendError::NotConnectedToPeer(peer_id))?
+            .ready()
+            .await
+            .map_err(|_| SendError::NotConnectedToPeer(peer_id))?
+            .send(data)
+            .map_err(|_| SendError::NotConnectedToPeer(peer_id))
```

Comment:
At least this error should be different than the two above, it means the connection has been pretty suddenly closed, which might be useful for debugging.